src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckJavaAgent.kt

```diff
@@ -404,7 +407,18 @@ internal object LincheckClassFileTransformer : ClassFileTransformer {
         val visitor = LincheckClassVisitor(writer, instrumentationMode, methods)
         try {
             reader.accept(visitor, ClassReader.EXPAND_FRAMES)
-            writer.toByteArray()
+            writer.toByteArray().also {
+                if (logTransformedSources) {
+                    val cr = ClassReader(it)
+                    val sw = StringWriter()
+                    val pw = PrintWriter(sw)
+                    cr.accept(org.objectweb.asm.util.TraceClassVisitor(pw), 0)
+
+                    File("build/transformed-by-lincheck/${classNode.name}.txt")
```

Comment:
Lets rename the folder into `build/instrumented-bytecode` or `build/transformed-bytecode`.

Reply:
It is not a problem at all, because this way the file system groups the classes in directories by package